aiida_workgraph/engine/workgraph.py

```diff
@@ -782,18 +782,28 @@ def run_tasks(self, names: t.List[str], continue_workgraph: bool = True) -> None
                         )
                     )
                     continue
+            # This calcfuntion/workfunction is already run
+            if (
+                task["metadata"]["node_type"].upper()
+                in [
+                    "CALCFUNCTION",
+                    "WORKFUNCTION",
+                ]
+                and self.get_task_state_info(name, "state").upper() != "PLANNED"
+            ):
+                continue
```

Comment:
This seems a bit hacky. Why is the work graph in a situation where it is inspecting a calcfunction/workfunction that has already run?

Reply:
Hi Edan, thanks for the comment. Yes, I replaced this by using `executed_tasks`. I also added a detailed description of the bug, and the proposed solution.